helm-charts/templates/data-plane/gateway-components/adapter/gateway.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: {{ .Values.wso2.apk.dp.gateway.name }}
+  annotations:
+    cert-manager.io/issuer: selfsigned-issuer
+spec:
+  gatewayClassName: {{ .Values.wso2.apk.dp.gateway.name }}
+  listeners:
+    - name: examplelistener
+      hostname: "*.example.com"
+      port: {{ .Values.wso2.apk.dp.gateway.listener.port }}
+      protocol: {{ .Values.wso2.apk.dp.gateway.listener.protocol }}
+      tls:
+        mode: Terminate
+        certificateRefs:
```

Comment:
Are we loading the cert/key pairs reading these cert refs?

Reply:
no, secrets are mounted and read from path, mounting secrets approach will be changed in the next iteration to read from cert refs